### PR TITLE
Subdirectives for path and listen address

### PR DESCRIPTION
**Pull request by MexHigh**
This PR will add the option to define the directive like this:
```
prometheus {
  address 0.0.0.0:1337
  path /metrics
}
```
or just like
```
prometheus 0.0.0.0:1337
```
to use the default path

### CODE REVIEW

**Comment by MexHigh** [comment]
I added tests for different prometheus directive usages. Don't know if they are useful. It's the first time I've written unittests in go.

File: go.mod [hunk 1]
```diff
@@ -4,6 +4,7 @@ go 1.14
 
 require (
 	github.com/caddyserver/caddy/v2 v2.1.1
+	github.com/caddyserver/xcaddy v0.1.5 // indirect
```
Comment (hairyhenderson) [inline go.mod]: I don't think this should be necessary to import?

File: caddyprom_test.go [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+
+	// metrics struct with default values
+	// TODO replace with actual metrics struct provisioned by caddy
+	metrics := Metrics{
+		Addr: "localhost:9180",
+		Path: "/metrics",
+	}
+
+	if err := metrics.UnmarshalCaddyfile(dispenser); err == nil {
+		t.Error("address can be set inline and as subdirective")
```
Comment (hairyhenderson) [inline caddyprom_test.go]: I think this should've been `can not`?

```suggestion
		t.Error("address can not be set inline and as subdirective")
```

File: caddyprom_test.go [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+			prometheus {
+				address 0.0.0.0:1337
+				path /otherpath
+			}
+		`,
+	}
+
+	// metrics struct with default values
+	// TODO replace with actual metrics struct provisioned by caddy
+	metrics := Metrics{
```
Comment (hairyhenderson) [inline caddyprom_test.go]: Shouldn't this be a `*Metrics`?

```suggestion
	metrics := &Metrics{
```

File: caddyprom_test.go [hunk 4]
```diff
@@ -0,0 +1,119 @@
...
```
Comment (hairyhenderson) [inline caddyprom_test.go]: Shouldn't this be a `*Metrics`?

```suggestion
	metrics := &Metrics{
```

**Comment by hairyhenderson** [comment]
Thanks @MexHigh - yes, the tests are useful, and give me some hint that they work 😉... I just realized that I never set this repo up for CI, so I'll have to manually test.

**Comment by hairyhenderson** [comment]
I'm going to add a couple changes to the tests, just to group them a bit more, and add a bit of coverage.

**Comment by hairyhenderson** [comment]
Thanks for the contribution @MexHigh!

**Pull request merged**